Resources/Prototypes/Recipes/Construction/Graphs/firelock.yml

```diff
@@ -0,0 +1,141 @@
+﻿- type: constructionGraph
+  id: Firelock
+  start: start
+  graph:
+    - node: start
+      edges:
+        - to: frame1
+          completed:
+            - !type:SnapToGrid { }
+          steps:
+            - material: Metal
+              amount: 3
+              doAfter: 1
+
+
+    - node: frame1
+      entity: FirelockFrame
+      edges:
+        - to: frame2
+          conditions:
+            - !type:EntityAnchored
+              anchored: true
+          steps:
+            - material: Cable
+              amount: 2
+              doAfter: 1
+        - to: start
+          completed:
+            - !type:SpawnPrototype
+              prototype: SteelSheet1
+              amount: 3
+            - !type:DeleteEntity {}
+          conditions:
+            - !type:EntityAnchored
+              anchored: true
+          steps:
+            - tool: Welding
+              doAfter: 1
+
+    - node: frame2
+      actions:
+        - !type:SpriteStateChange
+          state: frame2
+      edges:
+        - to: frame3
+          conditions:
+            - !type:EntityAnchored
+              anchored: true
+          steps:
+            - prototype: FirelockElectronics
+              icon:
+                sprite: Constructible/Misc/module.rsi
+                state: mainboard
+              name: Firelock Electronics
```

Comment:
Prototype steps are discouraged. Please make a "dummy" component called `FirelockElectronicsComponent`, give that to the firelock electronics prototype and use a component step instead.